File: app/src/main/java/com/aceplus/hackthon/login/LoginActivity.java
package com.aceplus.hackthon.login;

import android.content.Intent;
import android.os.Bundle;
import android.support.annotation.NonNull;
import android.support.v7.app.AppCompatActivity;
import android.text.TextUtils;
import android.view.View;
import android.widget.EditText;
import android.widget.LinearLayout;
import android.widget.Toast;

import com.aceplus.hackthon.MainActivity;
import com.aceplus.hackthon.R;
import com.google.android.gms.tasks.OnCompleteListener;
import com.google.android.gms.tasks.Task;
import com.google.firebase.auth.AuthResult;
import com.google.firebase.auth.FirebaseAuth;
import com.google.firebase.database.DatabaseReference;
import com.google.firebase.database.FirebaseDatabase;

import butterknife.BindView;
import butterknife.ButterKnife;
import butterknife.OnClick;
import butterknife.Optional;


public class LoginActivity extends AppCompatActivity{

    @BindView(R.id.progressLoading)
    LinearLayout progressBar;
    @BindView(R.id.edt_email)
    EditText edtEmail;
    @BindView(R.id.edt_password)
    EditText edtPassword;
    private FirebaseAuth auth;
    private DatabaseReference mDatabase;


    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_login);
        init();

    }

    private void init() {
        ButterKnife.bind(this);
        auth = FirebaseAuth.getInstance();
        mDatabase = FirebaseDatabase.getInstance().getReference();
        if (auth.getCurrentUser() != null) {
            startActivity(new Intent(LoginActivity.this, MainActivity.class));
            finish();
        }
    }

    @Optional
    @OnClick(R.id.btn_login)
    public void goToLogin() {

        final String email = edtEmail.getText().toString();
        final String password = edtPassword.getText().toString();

        if (TextUtils.isEmpty(email)) {
            Toast.makeText(getApplicationContext(), "Enter email address!", Toast.LENGTH_SHORT).show();
            return;
        }

        if (email.equalsIgnoreCase("@aceplussolutions.com")) {
            Toast.makeText(getApplicationContext(), "Just A User Name", Toast.LENGTH_SHORT).show();
            return;
        }

        if (TextUtils.isEmpty(password)) {
            Toast.makeText(getApplicationContext(), "Enter password!", Toast.LENGTH_SHORT).show();
            return;
        }
        progressBar.setVisibility(View.VISIBLE);
        String finalEmail = email+ "@aceplussolutions.com";
        auth.signInWithEmailAndPassword(finalEmail, password)
                .addOnCompleteListener(LoginActivity.this, new OnCompleteListener<AuthResult>() {
                    @Override
                    public void onComplete(@NonNull Task<AuthResult> task) {
                        if (!task.isSuccessful()) {
                            progressBar.setVisibility(View.GONE);
                            if (password.length() < 6) {
                                edtPassword.setError("Password too short, enter minimum 6 characters!");
                            } else {
                                Toast.makeText(LoginActivity.this, "Authentication failed.", Toast.LENGTH_LONG).show();
                            }
                        } else {
                            progressBar.setVisibility(View.GONE);
                            Intent intent = new Intent(LoginActivity.this, MainActivity.class);
                            startActivity(intent);
                            finish();
                        }
                    }
                });
    }

    @Optional
    @OnClick(R.id.txt_signUp)
    public void goToSignUp() {

        Intent i = new Intent(this, SignUpActivity.class);
        this.startActivity(i);
    }

    @Optional
    @OnClick(R.id.txt_forgotPassword)
    public void goToResetPassword() {

        Intent i = new Intent(this, ResetPasswordActivity.class);
        this.startActivity(i);
    }

    @Override
    public void onPointerCaptureChanged(boolean hasCapture) {

    }
}

File: backend/src/main/java/com/aceplus/backend/ui/activity/LoginActivity.kt
package com.aceplus.backend.ui.activity

import android.content.SharedPreferences
import android.os.Bundle
import android.preference.PreferenceManager
import android.support.v7.app.AppCompatActivity
import android.widget.Toast
import com.aceplus.backend.R
import kotlinx.android.synthetic.main.activity_login.*
import mvp.presenter.LoginPresenter
import mvp.view.LoginView

class LoginActivity : AppCompatActivity(), LoginView {
    private lateinit var preferences: SharedPreferences
    private lateinit var mPresenter: LoginPresenter
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_login)
        preferences = PreferenceManager.getDefaultSharedPreferences(applicationContext)
        if (preferences.getString("isLogin", "false") == "true") {
            navigateToMainActivity()
        }

        mPresenter = LoginPresenter(this)
        btnLogin.setOnClickListener { mPresenter.loginAdmin(edtUsername.text.toString(), edtPassword.text.toString()) }
    }

    override fun navigateToMainActivity() {
        preferences.edit().putString("isLogin", "true").apply()
        val intent = MainActivity.newIntent(applicationContext)
        startActivity(intent)
        finish()
    }

    override fun displayMessage(message: String) {
//        Toast.makeText(applicationContext, message, Toast.LENGTH_SHORT).show()
    }

}
